src/components/CompareResults/hashToCommitLoader.ts

```diff
@@ -0,0 +1,57 @@
+import { checkValues, getComparisonInformation } from './loader';
+import { fetchRevisionFromHash } from '../../logic/treeherder';
+import { Repository } from '../../types/state';
+
+// This function is responsible for fetching the data from the URL. It's called
+// by React Router DOM when the compare-results path is requested.
```

Comment:
Small nit, but you may want to update these comments here for the right context.

Reply:
You can use the old comments you had from the `loader` to explain why we need this `hashToCommitLoader`